framework/src/utils/cidr/v4.rs

```diff
@@ -49,6 +49,14 @@ impl Cidr for Ipv4Cidr {
     fn contains(&self, address: Self::Addr) -> bool {
         self.prefix == (self.mask & u32::from_be_bytes(address.octets()))
     }
+
+    #[inline]
+    fn contains_ip(&self, ip: IpAddr) -> bool {
```

Comment:
Not a fan of this. feel the abstraction is wrong. But we will revisit later.

Reply:
Yeah, I'm in general agreement, but seems useful enough for now, instead of changing a bunch of things to work across v4/v6 all the time.